docs/tutorials/text_prediction/automm_customization.md

```diff
@@ -0,0 +1,236 @@
+- `"uniform_soup"`: average all the top k checkpoints as the final checkpoint.
+- `"best"`: pick the checkpoint with the best validation performance.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"optimization.top_k_average_method": "greedy_soup"})
+# average all the top k checkpoints
+predictor.fit(hyperparameters={"optimization.top_k_average_method": "uniform_soup"})
+```
+
+### optimization.efficient_finetune
+Finetune only a small portion of parameters instead of one whole pretrained backbone.
+- `"bit_fit"`: bias parameters only.
+- `"norm_fit"`: normalization parameters + bias parameters.
+- `"lora"`: LoRA Adaptors. See [this paper](https://arxiv.org/abs/2106.09685) for details.
+- `"lora_bias"`: LoRA Adaptors + bias parameters.
+- `"lora_norm"`: LoRA Adaptors + normalization parameters + bias parameters.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"optimization.efficient_finetune": None})
+# finetune only bias parameters
+predictor.fit(hyperparameters={"optimization.efficient_finetune": "bit_fit"})
+```
+
+## Environment
+
+### env.num_gpus
+The number of gpus to use.
+```
+# by default, all available gpus are used by AutoMM
+predictor.fit(hyperparameters={"env.num_gpus": -1})
+# use 1 gpu only
+predictor.fit(hyperparameters={"env.num_gpus": 1})
+```
+
+### env.per_gpu_batch_size
+The batch size for each GPU. 
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.per_gpu_batch_size": 8})
+# use batch size 16 per GPU
+predictor.fit(hyperparameters={"env.per_gpu_batch_size": 16})
+```
+
+### env.batch_size
+The batch size to use in each step of training. If `env.batch_size` is larger than `env.per_gpu_batch_size * env.num_gpus`, we accumulate gradients to reach the effective `env.batch_size` before performing one optimization step.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.batch_size": 128})
+# use batch size 256
+predictor.fit(hyperparameters={"env.batch_size": 256})
+```
+
+### env.eval_batch_size_ratio
+Prediction or evaluation uses a larger per gpu batch size `env.per_gpu_batch_size * env.eval_batch_size_ratio`.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.eval_batch_size_ratio": 4})
+# use 2x per gpu batch size during prediction or evalution
+predictor.fit(hyperparameters={"env.eval_batch_size_ratio": 2})
+```
+
+### env.precision
+Support either double (`64`), float (`32`), bfloat16 (`"bf16"`), or half (`16`) precision training.
+
+Half precision, or mixed precision, is the combined use of 32 and 16 bit floating points to reduce memory footprint during model training. This can result in improved performance, achieving +3X speedups on modern GPUs.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.precision": 16})
+# use bfloat16
+predictor.fit(hyperparameters={"env.precision": "bf16"})
+```
+
+### env.num_workers
+The number of worker processes used by Pytorch dataloader in training.
```

Comment:
Can we write some suggestions here, like depending on your number of CPU cores or number of GPUs, what number of worker process should the user adopt to maximize throughput?

Reply:
Good suggestion.